### PR TITLE
Updating TPM readme

### PR DESCRIPTION
**Pull request by nickloestartup**
Updating the readme for Technical Planning Meetings.

This follows a review of the process and the need to have a greater consideration of the business impact following TPMs.

### CODE REVIEW

File: technical-planning-meeting.md [hunk 1]
```diff
@@ -3,39 +3,44 @@
...
-* 1 or more software architects
-* 1 or more database gurus (if database work will be required)
-* 1 or more UXUI gurus (if frontend work will be required)
-* 0 or more business owners (if the business requirements are sufficiently complex)
-* 0 or more testers (if there is a large testing requirement)
-* 0 or more project gurus (if developers might require additional technical assistance)
+## Probable Attendees
+
+* Engineer(s) with a technical challenge
+* Engineer(s) to support the disucssion and agree a technical solution/ next steps
```
Comment (HanCork89) [inline technical-planning-meeting.md]: discussion

File: technical-planning-meeting.md [hunk 2]
```diff
@@ -3,39 +3,44 @@
...
-* They should be requested by developers.
+* They should be requested by engineers.
 
 ## When is one appropriate?
 
 * When starting a new project
 * When adding new features or functionality
-* When undertaking anything that is likely to be more than 5 days work
+* When the technical implementation is not clear for a feature iteration
+* When undertaking work that requires multiple teams to collaborate on a techincal solution
```
Comment (HanCork89) [inline technical-planning-meeting.md]: technical

**Comment by HanCork89** [comment]
+1 from me

File: technical-planning-meeting.md (hunk 1 above)
Comment (nickloestartup) [reply technical-planning-meeting.md]: Done, thanks

File: technical-planning-meeting.md (hunk 2 above)
Comment (nickloestartup) [reply technical-planning-meeting.md]: Done, thanks

File: technical-planning-meeting.md [hunk 3]
```diff
@@ -3,39 +3,44 @@
 
 ## Definition
 
-* A technical planning meeting is an informal discussion aimed at streamlining our development processes.
-* They can be as short or as long as necessary.
-* They should be requested by developers.
+* A technical planning meeting (TPM) is an informal discussion aimed at streamlining our development processes and agreeing a technical solution
```
Comment (connormeredith) [inline technical-planning-meeting.md]: >and agreeing *on* a technical solution *and/ or next steps*?

File: technical-planning-meeting.md [hunk 4]
```diff
@@ -3,39 +3,44 @@
...
+* They can be as short or as long as necessary
+* They should be requested by engineers
 
 ## When is one appropriate?
 
 * When starting a new project
 * When adding new features or functionality
-* When undertaking anything that is likely to be more than 5 days work
+* When the technical implementation is not clear for a feature iteration
```
Comment (connormeredith) [inline technical-planning-meeting.md]: >for a feature *or* iteration?

File: technical-planning-meeting.md [hunk 5]
```diff
@@ -3,39 +3,44 @@
...
+* When undertaking work that requires multiple teams to collaborate on a technical solution
 
 ## Expected Outcomes
 
-* High level documentation of the business problem and technical solution
-* Developers should have a clear understanding of what is expected
-* A set of Jiras for product owners to prioritise
-* A definition of done
+* High level documentation of the business problem, the technical solution and/ or next steps and the business value (it is clear the complexity of the task in order that we can establish the business benefits)
```
Comment (connormeredith) [inline technical-planning-meeting.md]: >it is clear the complexity of the task in order that we can establish the business benefits

This doesn't make sense to me.

File: technical-planning-meeting.md [hunk 6]
```diff
@@ -3,39 +3,44 @@
...
+* Engineers should have a clear understanding of what is expected
```
Comment (connormeredith) [inline technical-planning-meeting.md]: >of what is expected *moving forward*?

File: technical-planning-meeting.md [hunk 7]
```diff
@@ -3,39 +3,44 @@
...
+* What the next steps are, this could be a set of Jiras for Pod Leads to prioritise 
+* A definition of done/ acceptance criteria
+* Pod Leads or Stakeholders should be clear what Tech Debt there is with any MVP (Minimum Viable Product)
```
Comment (connormeredith) [inline technical-planning-meeting.md]: Maybe a link to the [Tech Debt](https://github.com/holidayextras/culture/blob/master/tech-debt.md) documentation?

File: technical-planning-meeting.md [hunk 8]
```diff
@@ -3,39 +3,44 @@
...
-## Prerequisites
 
-* 1 or more developers with a technical challenge
-* 1 or more software architects
-* 1 or more database gurus (if database work will be required)
-* 1 or more UXUI gurus (if frontend work will be required)
-* 0 or more business owners (if the business requirements are sufficiently complex)
-* 0 or more testers (if there is a large testing requirement)
-* 0 or more project gurus (if developers might require additional technical assistance)
+## Probable Attendees
```
Comment (connormeredith) [inline technical-planning-meeting.md:26]: Should this also reflect the attendees info in the TPM document?
```
> If INF required, invite Damien Turner
> If data engineer required, ask in #pod-data for a rep
> If Multimedia required, invite Jess Everton
```

File: technical-planning-meeting.md [hunk 9]
```diff
@@ -3,39 +3,44 @@
...
 ## The Process
 
-1. The developers explain the business problem they're trying to solve and open it up for discussion.
-2. The developers explain their technical solution and open it up for discussion.
-3. We exchange thoughts and ideas, fill in any blanks and compromise where necessary to reach a good solution.
-4. We break the work down into small chunks.
-5. If the chunks of work are small enough to begin work, raise some Jiras. Otherwise, arrange for future technical planning meetings to dig deeper into the problem (an example of when this might be necessary - "We want to start selling Coach")
-6. The developers should leave the meeting and produce a markdown document covering everything discussed in parts #1 to #4
-7. Product owners prioritise the work as appropriate
+1. Make a copy of the [TPM template](https://docs.google.com/a/holidayextras.com/document/d/1zVbOz0dRAnzZ6UhjYO1Ce1YGI2dzvEoYvZfkTK7qjYU/edit?usp=sharing). Part 1 is completed and to increase visibility that the TPM is taking place, notification of the TPM is posted with a link to the document in the Slack Channel #tpm
```
Comment (connormeredith) [inline technical-planning-meeting.md]: This might be a good opportunity to group this into:
- Before
- During
- After

**Review by connormeredith — COMMENTED** [review]
Good stuff @nickloestartup, just a few comments :)

File: technical-planning-meeting.md (hunk 3 above)
Comment (nickloestartup) [reply technical-planning-meeting.md]: Agree, done.

File: technical-planning-meeting.md (hunk 4 above)
Comment (nickloestartup) [reply technical-planning-meeting.md]: I've reworded for clarity

File: technical-planning-meeting.md (hunk 5 above)
Comment (nickloestartup) [reply technical-planning-meeting.md]: I've reworded

File: technical-planning-meeting.md (hunk 6 above)
Comment (nickloestartup) [reply technical-planning-meeting.md]: Agree, done.

File: technical-planning-meeting.md (hunk 7 above)
Comment (nickloestartup) [reply technical-planning-meeting.md]: Agree, done.

File: technical-planning-meeting.md (hunk 8 above)
Comment (nickloestartup) [reply technical-planning-meeting.md:26]: Agree, done.

File: technical-planning-meeting.md (hunk 9 above)
Comment (nickloestartup) [reply technical-planning-meeting.md]: Agree, done.

**Comment by nickloestartup** [comment]
@connormeredith thanks for reviewing and the comments, really helpful :-)

I've updated, if you can have a 2nd look. Thanks

**Comment by nickloestartup** [comment]
@HanCork89 a few changes following Connor's review/ comments. If you want to take another look :-)

**Review by connormeredith — APPROVED** [review]
Great stuff :smile:

**Comment by HanCork89** [comment]
@connormeredith made some great observations so I'm more than happy to approve!

**Comment by nickloestartup** [comment]
@HanCork89 & @connormeredith 
Following a conversation with Steve C-S, I've made a small but important change. Steve argued _"I'd question what the complexity has to do with the [business] value, as something can be highly valuable but very simple."_ This is a really important point and after thought I've agreed that value should be replaced with impact. That is essentially what we're after, the impact, not the value - a high level indication of complexity and possible time to complete the work to understand the impact to the business of doing the work.

Just to make you aware of this change.

**Pull request merged**